File: backend/src/seed_admin.py
"""Idempotent seed of the initial super-admin account.

Creates a super-admin on first startup if none exists. Credentials are read
from env vars with safe dev defaults. CHANGE THESE before any public deploy.

Env overrides:
  ADMIN_USERNAME  (default: admin)
  ADMIN_PASSWORD  (default: Admin@1234)
  ADMIN_FULLNAME  (default: Super Admin)
"""
import os
from sqlalchemy import func
from .database import SessionLocal
from .models import AdminUser
from .auth import hash_password


def seed_super_admin():
    username = os.environ.get("ADMIN_USERNAME", "admin")
    password = os.environ.get("ADMIN_PASSWORD", "Admin@1234")
    fullname = os.environ.get("ADMIN_FULLNAME", "Super Admin")
    db = SessionLocal()
    try:
        exists = db.query(func.count(AdminUser.id)).filter(AdminUser.username == username).scalar()
        if exists:
            return
        db.add(AdminUser(
            username=username,
            full_name=fullname,
            password_hash=hash_password(password),
            role="super_admin",
            is_active=True,
        ))
        db.commit()
        print(f"[seed] super-admin '{username}' created (change password before production!)")
    finally:
        db.close()

File: _backup/php-backend/seed_admin.php
<?php
/**
 * Create the super-admin account on first deployment.
 * Run manually via:  php seed_admin.php
 */

require_once __DIR__ . '/includes/Database.php';
require_once __DIR__ . '/includes/Auth.php';
require_once __DIR__ . '/config.php';

// Check if admin already exists
$existing = Database::fetchOne(
    'SELECT id FROM admin_users WHERE username = ?',
    [ADMIN_USERNAME]
);

if ($existing) {
    echo "Admin '" . ADMIN_USERNAME . "' already exists (id={$existing['id']}). Skipping.\n";
    exit(0);
}

$hash = AuthHelper::hashPassword(ADMIN_PASSWORD);
Database::execute(
    'INSERT INTO admin_users (username, full_name, password_hash, role, is_active) VALUES (?, ?, ?, ?, 1)',
    [ADMIN_USERNAME, ADMIN_FULLNAME, $hash, 'super_admin']
);

echo "Super-admin '" . ADMIN_USERNAME . "' created (id=" . Database::lastInsertId() . ").\n";
echo "IMPORTANT: Change the admin password after first login!\n";
